akka-projection-testkit/src/main/scala/akka/projection/testkit/scaladsl/TestProjection.scala

```diff
@@ -0,0 +1,114 @@
+package akka.projection.testkit.scaladsl
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import akka.actor.typed.ActorSystem
+import akka.projection.StatusObserver
+import akka.projection.internal.ActorHandlerInit
+import akka.stream.scaladsl.Source
+import akka.NotUsed
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.stream.KillSwitches
+import akka.stream.SharedKillSwitch
+import akka.Done
+import akka.persistence.query.Offset
+import akka.projection.RunningProjection
+import akka.projection.internal.NoopStatusObserver
+import akka.projection.internal.SettingsImpl
+import akka.projection.internal.RestartBackoffSettings
+import akka.projection.scaladsl.Handler
+import akka.projection.scaladsl.SourceProvider
+import akka.projection.EventEnvelope
+
+// FIXME: this should be refactored as part of #198
+object TestProjection {
+  def apply[Event](
+      system: ActorSystem[_],
+      projectionId: ProjectionId,
+      sourceProvider: SourceProvider[Offset, EventEnvelope[Event]],
+      handler: Handler[EventEnvelope[Event]]): Projection[EventEnvelope[Event]] =
+    new TestProjection(projectionId, sourceProvider, handler)(system)
+}
+
+class TestProjection[Event](
+    val projectionId: ProjectionId,
+    sourceProvider: SourceProvider[Offset, EventEnvelope[Event]],
```

Comment:
I don't think we should base the TestProjection on EventEnvelope. It's specific to eventsByTag, with seqNr and persistenceId. I think it should be possible to have a TestProjection with a generic envelope type.